cadc-util/src/main/java/ca/nrc/cadc/db/version/KeyValueDAO.java

```diff
@@ -311,6 +347,22 @@ public Object extractData(ResultSet rs) throws SQLException {
         }
     }
 
+    private class KeyValueListExtractor implements ResultSetExtractor<List<KeyValue>> {
+
```

Comment:
This could/should be a RowMapper instead of a ResultSetExtractor. Then it could be used in the list() method and in the (poorly named) ModelVersionExtractor, which should be renamed to KeyValueExtractor. The reason that one is a ResultSetExtractor is that it simplifies the get() and lock() methods by returning a single object. RowMapper is generally quite re-usable.

Reply:
also, RowMapper can be used with `queryForObject` for a type-safe single-row return value; ResultSetExtractor is only needed if the returned object spans multiple rows in the result set.

Reply:
Updated to use a RowMapper.